rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -32,58 +34,108 @@
 
 public class RegisterBtcTransactionIT {
 
+    public static final Coin BTC_TRANSFERRED = Coin.COIN;
     private final BridgeConstants bridgeConstants = BridgeMainNetConstants.getInstance();
     private final NetworkParameters btcParams = bridgeConstants.getBtcParams();
     private final BridgeSupportBuilder bridgeSupportBuilder = BridgeSupportBuilder.builder();
-
-    @Test
-    void whenRegisterALegacyBtcTransaction_shouldRegisterTheNewUtxoAndTransferTheRbtcBalance() {
-        // Arrange
-        ActivationConfig.ForBlock activationConfig = ActivationConfigsForTest.all().forBlock(0);
-        Repository repository = BridgeSupportTestUtil.createRepository();
-        Repository track = repository.startTracking();
-        Block rskExecutionBlock = getRskExecutionBlock();
-
-        BtcLockSenderProvider btcLockSenderProvider = new BtcLockSenderProvider();
-        FeePerKbSupport feePerKbSupport = getFeePerKbSupport();
+    private ActivationConfig.ForBlock activationConfig;
+    private Repository track;
+    private Repository repository;
+    private Block rskExecutionBlock;
+    private BtcLockSenderProvider btcLockSenderProvider;
+    private FeePerKbSupport feePerKbSupport;
+    private FederationSupport federationSupport;
+    private BridgeStorageProvider bridgeStorageProvider;
+    private BtcBlockStoreWithCache.Factory btcBlockStoreFactory;
+    private BtcBlockStoreWithCache btcBlockStoreWithCache;
+    private BtcTransaction bitcoinTransaction;
+    private PartialMerkleTree pmtWithTransactions;
+    private int btcBlockWithPmtHeight;
+    private int chainHeight;
+    private Transaction rskTx;
+    private ECKey ecKey;
+    private RskAddress rskReceiver;
+    private BridgeSupport bridgeSupport;
+
+    @BeforeEach
+    void setUp() {
+        rskTx = TransactionUtils.createTransaction();
+        repository = BridgeSupportTestUtil.createRepository();
+        track = repository.startTracking();
+
+        activationConfig = ActivationConfigsForTest.all().forBlock(0);
+        rskExecutionBlock = getRskExecutionBlock();
+
+        btcLockSenderProvider = new BtcLockSenderProvider();
+        feePerKbSupport = getFeePerKbSupport();
 
         Federation federation = P2shErpFederationBuilder.builder().build();
         FederationStorageProvider federationStorageProvider = getFederationStorageProvider(track, federation);
-        FederationSupport federationSupport = getFederationSupport(federationStorageProvider, activationConfig, bridgeConstants.getFederationConstants());
+        federationSupport = getFederationSupport(federationStorageProvider, activationConfig, bridgeConstants.getFederationConstants());
```

Comment:
I would recommend to just use the builder here. No need to create a private method for this. Same for the other private methods where a builder is being used. This will reduce more boilerplate code.